ext/session/session.c

```diff
@@ -2344,15 +2782,64 @@ static PHP_FUNCTION(session_start)
 }
 /* }}} */
 
-/* {{{ proto bool session_destroy(void)
+
+/* {{{ proto array session_info(void)
+   Return session internal data information */
+static PHP_FUNCTION(session_info)
+{
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (PS(session_status) != php_session_active) {
+		php_error_docref(NULL, E_WARNING, "Session is not active");
+		RETURN_FALSE;
+	}
+
+	ZEND_ASSERT(!Z_ISUNDEF(PS(internal_data)));
+	ZVAL_COPY(return_value, &PS(internal_data));
+	return;
+}
+/* }}} */
+
+
+/* {{{ proto int session_gc(void)
+   Perform session GC manually */
+static PHP_FUNCTION(session_gc)
+{
+	zend_long nrdels;
+
+	if (zend_parse_parameters_none() == FAILURE) {
+		return;
+	}
+
+	if (PS(session_status) != php_session_active) {
+		php_error_docref(NULL, E_WARNING, "Session is not active");
+		RETURN_FALSE;
+	}
+
+	nrdels = php_session_gc();
+	if (nrdels < 0) {
+		RETURN_FALSE;
+	}
+	ZVAL_LONG(return_value, nrdels);
+}
+/* {{{ proto bool session_destroy([bool immediate])
    Destroy the current session and all data associated with it */
 static PHP_FUNCTION(session_destroy)
 {
-	if (zend_parse_parameters_none() == FAILURE) {
+	zend_bool immediate=0;
```

Comment:
While the intent might be fine (if this were a new function), BC breaking this is a potential security problem.
E.g. you have detected a breach and want to invalidate a session. Now, old code is just using `session_destroy()` and attacker can still use it.

I strongly urge you to use true by default. It's better to randomly rarely loose sessions than having actual sessions leaking to an attacker.